### DISCUSSION TITLE
How does --env-name work (if it does) ?

### DISCUSSION THREAD

**Discussion by dobesv** [swc-project/swc#5423] [Q&A]
I see mention in the docs of `--env-name` but I can't quite tell how it is applied to the configuration.  Is this a work in progress, or just lacking documentation?

**Comment by houyaowei** [comment]
I got the same problem.

`
 npx swc ./src -d dist --include-dotfiles --env-name='test'

`


when cliOptions is print in console, but env config is not found:

 
cliOptions: {
  outDir: 'dist',
  outFile: undefined,
  stripLeadingPaths: false,
  filename: undefined,
  filenames: [ './src' ],
  sync: false,
  workers: undefined,
  sourceMapTarget: undefined,
  extensions: [
    '.js',  '.jsx',
    '.es6', '.es',
    '.mjs', '.ts',
    '.tsx', '.cts',
    '.mts'
  ],
  watch: false,
  copyFiles: false,
  outFileExtension: 'js',
  includeDotfiles: true,
  deleteDirOnStart: false,
  quiet: false,
  only: [],
  ignore: []
}

**Reply by houyaowei** [reply]
it seems not working properly

**Reply by kdy1** [reply]
Do you have `.env.test`?

**Reply by kdy1** [reply]
I'm not sure what's `--env-name`, though. I copied CLI from babel

**Reply by houyaowei** [reply]
<img width="198" alt="截屏2025-01-09 15 56 19" src="https://github.com/user-attachments/assets/d68678be-a81f-4b48-acc7-6d3779c89e8d" />
